Allow SimpleLiteral to be a vararg

TheLimeGlass opened the pull request:
Allow SimpleLiteral to be a vararg for accepting a single value.

TheLimeGlass committed:
Allow SimpleLiteral to be a vararg

Allow SimpleLiteral to be a vararg

Mwexim commented on src/main/java/io/github/syst3ms/skriptparser/lang/SimpleLiteral.java:
```diff
@@ -19,11 +19,12 @@
+    private final T[] values;
 
-    public SimpleLiteral(T[] values) {
+    public SimpleLiteral(T... values) {
```
Won't this create issues for SyntaxParser at line 538? Those lines over there should be changed as well. Also, should the other constructor be removed as well or can it make a difference? I'm not experienced enough with Java generics to know if that constructor should exist or not (because it already has a varargs parameter).

TheLimeGlass replied:
It shouldn't, but it'll remove that whole Array newInstance. Epic. I will get around to changing that soon.

Mwexim committed:
refactor: changed SimpleLiteral use-cases to allow varargs in its constructor

Mwexim committed:
fix: reverted original changes but still left some changes to the SimpleLiteral

Vararg generics are just not the way to go in Java and the code was not working when you'd use it as expected. The original changes were therefore reverted, but there was still a small change left in the SimpleLiteral class.

Mwexim commented:
Sorry to hijack this pull request like this. My original intention was to just add the changes I proposed, but then I saw that the proposal itself is just not a good idea because of the way varargs and generics are handled (there's a reason the original author Syst3ms) added a specific constructor that needed a class argument as well).

That said, I still made some minor changes, hence why I'm still merging the pull request.
